eng/packages/http-client-csharp-mgmt/generator/Azure.Generator.Management/src/ManagementInputLibrary.cs

```diff
@@ -46,16 +44,24 @@ public ManagementInputLibrary(string configPath) : base(configPath)
         internal bool IsResourceModel(InputModelType model)
             => model.Decorators.Any(d => d.Name.Equals(KnownDecorators.ArmResourceInternal));
 
-        private IReadOnlyList<InputClient> EnumerateClients()
+        internal bool IsResourceModelType(CSharpType type) => ResourceTypes.Contains(type);
+
+        private HashSet<CSharpType> BuildResourceModels()
         {
-            var clients = new List<InputClient>(InputNamespace.Clients);
-            for (int i = 0; i < clients.Count; i++)
+            var resourceTypes = new HashSet<CSharpType>();
+
+            foreach (var model in InputNamespace.Models)
             {
-                var client = clients[i];
-                clients.AddRange(client.Children);
+                if (IsResourceModel(model))
+                {
+                    var modelProvider = ManagementClientGenerator.Instance.TypeFactory.CreateModel(model);
+                    if (modelProvider is not null)
+                    {
+                        resourceTypes.Add(modelProvider.Type);
+                    }
+                }
             }
-
-            return clients;
+            return resourceTypes;
```

Comment:
having `CSharpType` in it makes this no longer a "input" component.
We could consider to move this to output, or we could use the csharptype to provider dict to know whether a CSharpType is a resource data.